pkg/fqdn/namemanager/gc.go

```diff
@@ -173,28 +173,19 @@ func (n *manager) doGC(ctx context.Context) error {
 	return nil
 }
 
-func (n *manager) StartGC(ctx context.Context) {
-	n.manager.UpdateController(dnsGCJobName, controller.ControllerParams{
-		Group:       dnsGCControllerGroup,
-		RunInterval: DNSGCJobInterval,
-		DoFunc:      n.doGC,
-		Context:     ctx,
-	})
-}
-
 // RestoreCache loads cache state from the restored system:
 // - adds any pre-cached DNS entries
 // - repopulates the cache from the (persisted) endpoint DNS cache and zombies
-func (n *manager) RestoreCache(preCachePath string, eps map[uint16]*endpoint.Endpoint) {
+func (n *manager) RestoreCache(eps map[uint16]*endpoint.Endpoint) {
 	// Prefill the cache with the CLI provided pre-cache data. This allows various bridging arrangements during upgrades, or just ensure critical DNS mappings remain.
-	if preCachePath != "" {
+	if n.params.Config.ToFQDNsPreCache != "" {
 		n.logger.Info("Reading toFQDNs pre-cache data")
-		precache, err := readPreCache(preCachePath)
+		precache, err := readPreCache(n.params.Config.ToFQDNsPreCache)
 		if err != nil {
 			// FIXME: add a link to the "documented format"
```

Comment:
so, uh, can we add a link to the "documented format" as a drive-by? Or is it just undocumented? I couldn't find the link 😅

Reply:
Aha, even easier. It was added for upgrade from v1.3 to v1.4. We can kill it :-)

Reply:
In the interest of making this PR any larger, I'll file a follow-up PR to remove the precache entirely.